src/main/java/uk/gov/hmcts/reform/sscs/helper/AppealPostcodeHelper.java

```diff
@@ -0,0 +1,39 @@
+package uk.gov.hmcts.reform.sscs.helper;
+
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Address;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Appellant;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Appointee;
+import uk.gov.hmcts.reform.sscs.validators.PostcodeValidator;
+
+@Component
+@RequiredArgsConstructor
+public class AppealPostcodeHelper {
+
+    private final PostcodeValidator postcodeValidator;
+
+    public String resolvePostcode(Appellant appellant) {
+
+        if (appellant == null) {
+            return StringUtils.EMPTY;
+        }
+
+        return Optional.ofNullable(appellant.getAppointee())
+            .map(Appointee::getAddress)
+            .map(Address::getPostcode)
+            .filter(this::isValidPostcode)
+            .orElse(Optional.ofNullable(appellant.getAddress())
+                .map(Address::getPostcode)
+                .filter(this::isValidPostcode)
+                .orElse(StringUtils.EMPTY));
```

Comment:
Should only be using appointee's postcode is isAppointee is Yes

Reply:
I don't trust bulk scan enough to have this set. If an appointee is provided in the exception record then I think it's safe to assume that this is the case. There's a related ticket to look into and fix the appointee issues in bulk scan